torch/_utils.py

```diff
@@ -266,3 +266,13 @@ def _take_tensors(tensors, size_limit):
     for buf, _ in buf_dict.values():
         if len(buf) > 0:
             yield buf
+
+
+# annotation decorator to get annotations in a way that is compatible
+# with both Python 2 and 3
+def annotate(ret, **kwargs):
```

Comment:
@zdevito Now here's a kind of interesting way for JIT to support type annotations in Python 2 :)

Reply:
Yeah, my main concern was "if this worked, the python people would have used that for Python 2 instead of the comments", but apparently not.

Reply:
Developers are not perfectly efficient! :)

Reply:
Well that might not work with mypy though. It's probably based on some static analysis/abstract interpretation, so it might not register those correctly.

Reply:
It would seem not that much harder to find the annotations in the AST (in contrast to Python 2 type comments...).
Happily, we have the pyi after we're done.